### PR TITLE
Feature/swagger docs

### LINKED ISSUE #73: Docs for endpoints
Use Swagger

### PR DESCRIPTION
**Pull request by nmlinaric**
Added Swagger documentation. Documented endpoints, models and model responses - the last two are mainly to simplify documenting possible future changes in model layouts.

Resolves #73 Docs for endpoints.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,7 +24,17 @@ services:
...
+  swagger:
+    image: swaggerapi/swagger-ui
+    container_name: "swagger_docs"
+    environment:
+      BASE_URL: /api/swagger
+      SWAGGER_JSON: /swagger.yaml
+    volumes:
+      - "./swagger.yaml:/swagger.yaml"
+    ports:
+      - "8080:8080"
```
Comment (morrigan) [inline docker-compose.yml]: 8080 on the left side should be configurable using ENV variable (check above for database)

File: docker-compose.yml [hunk 2]
```diff
@@ -24,7 +24,17 @@ services:
     - POSTGRES_DB=${DB_NAME:-nodefactory}
     ports:
     - ${DB_LOCAL_PORT:-9999}:5432
+  swagger:
```
Comment (morrigan) [inline docker-compose.yml]: I think you are missing command how to run this  in taskfile

Comment (nmlinaric) [reply docker-compose.yml]: The docs can be accessed on ```app-url:port/api/swagger``` when the app is ran. Wasn't sure if README.md was the appropriate place to write that down or something else?

Comment (morrigan) [reply docker-compose.yml]: yep :)

Also, should backend docker image then depend on this?

**Comment by nmlinaric** [comment]
> Have you tried swagger api calls?
> I think they will run against swagger url by default. You should specify api/server url somewhere and it should be configurable by env variable.

Didn't know about that one (if you mean this: https://inspector.swagger.io/builder), but It's crashing  when I try to save a response. Nvm, maybe next time I'll need it, it will work :) 
I'll add the possibility to configure the url via env variable.

File: README.md [hunk 3]
```diff
@@ -30,11 +30,18 @@ Run `npx task --help` to see all available commands and their description.
 * use sequelize-cli local to generate migrations (because of timestamp)
 
 
+## Documentation
+
+The documentation can be accessed on `app-url:docs-port/docs-url`
```
Comment (morrigan) [inline README.md]: Put example here  -> i.e. `http://localhost:8080/api/swagger/` as these don't look like variables really

File: docs/types/miningRewards.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            type: string
+responses:
+  MiningRewards:
+    type: object
+    properties:
+      id:
+        type: integer
+      cid:
+        type: string
+      reward:
```
Comment (morrigan) [inline docs/types/miningRewards.yaml]: I think this one is called `totalSum` (I remember because I hated to change it now on frontend :D )

**Comment by morrigan** [comment]
@mpetrunic can you approve? :D

File: README.md (hunk 3 above)
Comment (mpetrunic) [reply README.md]: Wait something is wrong here. put here ${ENV_VARIABLE} like ${SWAGGER_DOCS_URL}

File: .env.sample [hunk 5]
```diff
@@ -37,3 +37,9 @@ EMAIL_UPTIME_NOTIF_TEMPLATE_ID=2
 # Cron like sintax
 UPTIME_NOTIF_RECURRENCE=0 * * * *
 
+# Swagger settings
+
+# Swagger docs default port
+SWAGGER_DOCS_PORT=8080
+# Swagger docs default url
+SWAGGER_DOCS_URL=/api/swagger
```
Comment (mpetrunic) [inline .env.sample]: How do I define location of hactar-backend for api calls. From what I see this just define path on which swagger-ui is exposed and port. But by default swagger ui will invoke api on his root which when deployed won't be same as hactar-backend url

File: docker-compose.yml [hunk 6]
```diff
@@ -26,7 +26,18 @@ services:
...
+    image: swaggerapi/swagger-ui
+    container_name: "swagger_docs"
+    environment:
+      BASE_URL: "${SWAGGER_DOCS_URL:-/api/swagger}"
+      SWAGGER_JSON: /tmp/swagger.yaml
+    volumes:
+      - "./docs:/tmp"
```
Comment (mpetrunic) [inline docker-compose.yml]: - use /opt instead of /tmp

File: docker-compose.yml [hunk 7]
```diff
@@ -26,7 +26,18 @@ services:
...
+      - "./docs/types:/usr/share/nginx/html/types"
```
Comment (mpetrunic) [inline docker-compose.yml]: - "./docs:/tmp:ro"
- "./docs/types:/usr/share/nginx/html/types:ro"

will mark files as read only

**Pull request merged**